util/mcf_dict_util.py

```diff
@@ -0,0 +1,537 @@
+        is_match = True
+        for prop in prop_list:
+            if prop not in node_dict:
+                is_match = False
+        if is_match:
+            s = get_dcid_node(node_dict)
+            if s:
+                ret_list.append(s)
+    return ret_list
+
+
+# list dupe
+def node_list_check_existence_dc(node_list: list) -> dict:
+    """Checks the existence of dcid of each node in DC.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        
+    Returns:
+        Dict object with dcids as key values and boolean values signifying existence as values.
+    """
+    dcid_list = get_dcids_node_list(node_list)
+    return dc_check_existence(dcid_list)
+
+
+def node_list_check_existence_node_list(node_list: list,
+                                        master_list: list) -> dict:
+    """Checks the existence of dcid of each node in another list of nodes.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        master_list: List of OrderedDict or Dict objects representing a node in MCF file to compare against.
+        
+    Returns:
+        Dict object with dcids as key values and boolean values signifying existence as values.
+    """
+    ret_dict = {}
+    dcid_list = get_dcids_node_list(node_list)
+    master_dcid_list = get_dcids_node_list(master_list)
+    for dcid in dcid_list:
+        if dcid in master_dcid_list:
+            ret_dict[dcid] = True
+        else:
+            ret_dict[dcid] = False
+    return ret_dict
+
+
+# drop dcid list
+def drop_nodes(node_list: list, dcid_list: list) -> list:
+    """Drops the node from list of nodes if it's dcid is present in the dcid list.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        dcid_list: List of dcids to be dropped from the list.
+        
+    Returns:
+        List of OrderedDict or Dict objects representing a node in MCF file after dropping.
+    """
+    ret_list = []
+    for node_dict in node_list:
+        if not get_dcid_node(node_dict) in dcid_list:
+            ret_list.append(node_dict)
+    return ret_list
+
+
+def add_path_exp_mcf_dicts(new_path: str,
+                           existing_dict: Optional[dict] = None,
+                           reopen: bool = False) -> dict:
+    """Opens a set of files specified by `new_path` expression to create a list of nodes for each file.
+
+    Args:
+        new_path: Glob like path to open a set of files and create a list of node for each file.
+        existing_dict: (Optional) dict returned from previous call to add more files to same dict object.
+        reopen: Boolean value to skip opening files already present in the dictionary to prevent overwrites.
+        
+    Returns:
+        A dictionary with filenames as objects and a list of OrderedDict objects where each object 
+            represents a node in the MCF file as value.
```

Comment:
Pls add that the key is the file name and value is dictionary of nodes.

Reply:
fixed.